src/Common/MemoryTracker.cpp

```diff
@@ -104,10 +111,8 @@ void MemoryTracker::alloc(Int64 size)
     std::bernoulli_distribution fault(fault_probability);
     if (unlikely(fault_probability && fault(thread_local_rng)))
     {
-        free(size);
```

Comment:
This code was incorrect as we don't need to recursively subtract from the chain of memory trackers here (we did not increased the memory tracked by parent memory tracker).